shared/cache.ts

```diff
@@ -0,0 +1,47 @@
+import { DateTime } from "luxon";
+
+export type Cache<DataType> = {
+  value?: DataType;
+  updatedAt?: string;
+  timeToLive: number;
+};
+
+export const createCache = <DataType>(
+  defaultCache: Partial<Cache<DataType>> = {}
+) => {
+  let cache: Cache<DataType> = {
+    value: defaultCache.value ?? undefined,
+    updatedAt: defaultCache.updatedAt ?? undefined,
+    timeToLive: 5 * 60,
+  };
+
+  const getOrUpdateCache = async (getValue: () => Promise<DataType>) => {
+    if (isCacheExpired(cache)) {
+      try {
+        const value = await getValue();
+        cache = {
+          value,
+          updatedAt: DateTime.utc().toISO(),
+          timeToLive: cache.timeToLive,
+        };
+      } catch (e) {
+        console.error(`Error updating cache: ${e}`);
+      }
+    }
+
+    return cache.value;
```

Comment:
Strategy is to return the cached value, even on failure - this is baked into the cache here. Not sure if we would want that in every case?